### PR TITLE
Upgraded samples for OAuth samples

### PR DESCRIPTION
**Pull request by HariRangarajan-Solace**
- Check for expiration, audience, issuer claims to ensure token integrity.
- Implement logging and monitoring for all token-related activities.

### CODE REVIEW

File: src/main/java/com/solace/samples/jcsmp/snippets/HowToConfigureSessionUsingOAUTH2.java [hunk 1]
```diff
@@ -16,55 +16,106 @@
 
 package com.solace.samples.jcsmp.snippets;
 
-import com.solacesystems.jcsmp.JCSMPProperties;
+import com.solacesystems.jcsmp.*;
```
Comment (Mrc0113) [inline src/main/java/com/solace/samples/jcsmp/snippets/HowToConfigureSessionUsingOAUTH2.java:19]: did best practices change? shouldn't we be listing out the individual packages?

File: src/main/java/com/solace/samples/jcsmp/snippets/HowToConfigureSessionUsingOpenIdConnect.java [hunk 2]
```diff
@@ -16,61 +16,117 @@
 
 package com.solace.samples.jcsmp.snippets;
 
-import com.solacesystems.jcsmp.JCSMPProperties;
+import com.solacesystems.jcsmp.*;
```
Comment (Mrc0113) [inline src/main/java/com/solace/samples/jcsmp/snippets/HowToConfigureSessionUsingOpenIdConnect.java:19]: did best practices change? shouldn't we be listing out the individual packages?

File: src/main/java/com/solace/samples/jcsmp/snippets/HowToConfigureSessionUsingOAUTH2.java [hunk 3]
```diff
@@ -16,55 +16,106 @@
...
+    // remember to add log4j2.xml to your classpath
+    private static final Logger logger = LogManager.getLogger(); // log4j2, but could also use SLF4J, JCL, etc.
+
+    public void createOauth2Session(JCSMPProperties sessionProperties, String accessToken, String issuerId)
+            throws InvalidPropertiesException {
+        sessionProperties.setProperty(JCSMPProperties.AUTHENTICATION_SCHEME, JCSMPProperties.AUTHENTICATION_SCHEME_OAUTH2);
+        sessionProperties.setProperty(JCSMPProperties.OAUTH2_ACCESS_TOKEN, accessToken);
+        sessionProperties.setProperty(JCSMPProperties.OAUTH2_ISSUER_IDENTIFIER, issuerId);
+
+        // When the Access token has expired, the client will be disconnected, and
```
Comment (Mrc0113) [inline src/main/java/com/solace/samples/jcsmp/snippets/HowToConfigureSessionUsingOAUTH2.java:41]: shouldn't we be checking to see if the token is close to being expired and refreshing it before expiry?

File: src/main/java/com/solace/samples/jcsmp/snippets/HowToConfigureSessionUsingOpenIdConnect.java [hunk 4]
```diff
@@ -16,61 +16,117 @@
...
+    // remember to add log4j2.xml to your classpath
+    private static final Logger logger = LogManager.getLogger(); // log4j2, but could also use SLF4J, JCL, etc.
+
+    public void createOIDCSession(JCSMPProperties sessionProperties, String idToken, String accessTokenOptional, String issuerId) throws InvalidPropertiesException {
+        sessionProperties.setProperty(JCSMPProperties.AUTHENTICATION_SCHEME, JCSMPProperties.AUTHENTICATION_SCHEME_OAUTH2);
+        sessionProperties.setProperty(JCSMPProperties.OAUTH2_ACCESS_TOKEN, accessTokenOptional);
+        sessionProperties.setProperty(JCSMPProperties.OIDC_ID_TOKEN, idToken);
+
+        // When the OAuth token has expired, the client will be disconnected, and
+        // will try to reconnect. Refresh the token before it reconnects.
```
Comment (Mrc0113) [inline src/main/java/com/solace/samples/jcsmp/snippets/HowToConfigureSessionUsingOpenIdConnect.java:41]: shouldn't we be checking to see if the token is close to being expired and refreshing it before expiry?

File: src/main/java/com/solace/samples/jcsmp/snippets/HowToConfigureSessionUsingOAUTH2.java [hunk 5]
```diff
@@ -1,12 +1,12 @@
 /*
  * Copyright 2021-2022 Solace Corporation. All rights reserved.
```
Comment (Mrc0113) [inline src/main/java/com/solace/samples/jcsmp/snippets/HowToConfigureSessionUsingOAUTH2.java:2]: update year to be 2021-2024

File: src/main/java/com/solace/samples/jcsmp/snippets/HowToConfigureSessionUsingOpenIdConnect.java [hunk 6]
```diff
@@ -1,12 +1,12 @@
 /*
  * Copyright 2021-2022 Solace Corporation. All rights reserved.
```
Comment (Mrc0113) [inline src/main/java/com/solace/samples/jcsmp/snippets/HowToConfigureSessionUsingOpenIdConnect.java:2]: update year to be 2021-2024

File: src/main/java/com/solace/samples/jcsmp/snippets/HowToConfigureSessionUsingOpenIdConnect.java [hunk 7]
```diff
@@ -16,61 +16,117 @@
...
+        if (!validateTokenClaims(newAccessToken, issuerId)) {
+            throw new Exception("OAuth Token validation failed");
+        }
+        logger.info("OAuth Token successfully refreshed and validated");
+        return newAccessToken;
+    }
+
+    private String refreshOIDCAccessToken(String issuerId) throws Exception {
+        String newIdToken = fetchNewOidcIdToken();
+        if (newIdToken == null || newIdToken == null) {
```
Comment (Mrc0113) [inline src/main/java/com/solace/samples/jcsmp/snippets/HowToConfigureSessionUsingOpenIdConnect.java:74]: same check?

**Review by Mrc0113 — CHANGES_REQUESTED** [review]

File: src/main/java/com/solace/samples/jcsmp/snippets/HowToConfigureSessionUsingOpenIdConnect.java (hunk 4 above)
Comment (HariRangarajan-Solace) [reply src/main/java/com/solace/samples/jcsmp/snippets/HowToConfigureSessionUsingOpenIdConnect.java:41]: No, the new token will be pickedup only when the client tries to reconnect after an expired token exception.
So, the best practice as per the field (SEs and PSGs) is to catch the exception and handle the token refresh accordingly.

File: src/main/java/com/solace/samples/jcsmp/snippets/HowToConfigureSessionUsingOpenIdConnect.java (hunk 2 above)
Comment (HariRangarajan-Solace) [reply src/main/java/com/solace/samples/jcsmp/snippets/HowToConfigureSessionUsingOpenIdConnect.java:19]: yes, that is the best practice, but at somestage the IntelliJ code intelligence switches from multiple classes in the same package structure to * imports.

File: src/main/java/com/solace/samples/jcsmp/snippets/HowToConfigureSessionUsingOAUTH2.java (hunk 1 above)
Comment (HariRangarajan-Solace) [reply src/main/java/com/solace/samples/jcsmp/snippets/HowToConfigureSessionUsingOAUTH2.java:19]: Multiple classes from the same package impacts readability. Hence IntelliJ code completion compresses them to a single * based import

File: src/main/java/com/solace/samples/jcsmp/snippets/HowToConfigureSessionUsingOAUTH2.java (hunk 3 above)
Comment (HariRangarajan-Solace) [reply src/main/java/com/solace/samples/jcsmp/snippets/HowToConfigureSessionUsingOAUTH2.java:41]: Same reply as the other comment.
Once connected and authenticated from the broker, the new token will be picked only when the current token is expired and invalidated and an exception is thrown. 
As per PSGs and SEs, the recommended way is to wait for the disconnect and catch the exception and then provide a new token to try and connect.

**Pull request merged**